src/nupic/types/Sdr.hpp

```diff
@@ -506,6 +508,23 @@ class SparseDistributedRepresentation : public Serializable
      */
     void load(std::istream &inStream) override;
 
+		CerealAdapter;
+    template<class Archive>
+    void save_ar(Archive & ar) const
+    {
+        getSparse(); // to make sure sparse is valid.
+        ar(cereal::make_nvp("SDR", dimensions_), cereal::make_nvp("sparse", sparse_) );
+    }
+
+    template<class Archive>
+    void load_ar(Archive & ar)
+    {
+        setSparseInplace(); // to make sure sparse is valid.
```

Comment:
does this have to be here? 
And makes sense? (on an empty SDR) in load()

Reply:
..the one below is correct

Reply:
> does this have to be here?

That I was asking if this is correct.  But looking at his load( ) function for the other serialization it appears that this leading setSparseInplace() is not needed.